core/src/main/java/org/web3j/protocol/websocket/WebSocketService.java

```diff
@@ -114,22 +120,33 @@ private void connectToWebSocket() throws InterruptedException, ConnectException
         }
     }
 
-    private void setWebSocketListener() {
+    private void setWebSocketListener(Consumer<String> onMessage, Consumer<Throwable> onError,
+                                      Consumer<Void> onClose) throws ConnectException {
```

Comment:
onClose can be a Runnable?

Reply:
> onClose can be a Runnable?

@zacscoding Can we use Runnable?

Reply:
@olimasaidov
Sure :)
why requested pull again ?

Reply:
@zacscoding your pull request was merged into 5.x branch. We [need this functionality](https://github.com/web3j/web3j/pull/787#issuecomment-521595515) in 4.x versions.